File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int n;
    cin >> n;

    vector<int> num(n);
    for (int i = 0; i < n; i++)
    {
        cin >> num[i];
    }

    int x;
    cin >> x;

    sort(num.rbegin(), num.rend());

    int cnt = 0;
    int l = 0, r = n - 1;
    while (l < r)
    {
        if (num[l] + num[r] == x)
        {
            cnt++;
            l++;
            r--;
        }
        else if (num[l] + num[r] > x)
        {
            l++;
        }
        else
        {
            r--;
        }
    }

    cout << cnt;
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
n = int(input())
num = list(map(int, input().split()))
x = int(input())

num.sort(reverse=True)


cnt = 0
l = 0
r = n - 1
while l < r:
    if num[l] + num[r] == x:
        cnt += 1
        l += 1
        r -= 1
    elif num[l] + num[r] > x:
        l += 1
    else:
        r -= 1

print(cnt)
